src/fmu/dataio/dataio.py

```diff
@@ -109,8 +115,51 @@ def _check_global_config(
             "One or more keys required for valid metadata are not found: "
             f"{missing_keys} (perhaps the config is empty?) "
         )
+
+    # check "stratigraphy"
+    if "stratigraphy" in globalconfig:
+        # we currently allow (why?) stratigraphy key missing from config.
+        strat = globalconfig["stratigraphy"]
+        if not isinstance(strat, dict):
+            msg += "The 'stratigraphy' must be a dictionary.\n"
+
+        # Loop the entries in 'stratigraphy'
+        # These keys are custom, but we want error messages to point to the key when
+        # issues are discovered. This makes it tricky to use jsonschema. Pydantic might
+        # be an option. But for now, just go through all items and do defined checks.
+
+        for key, item in strat.items():
+            if "name" not in item:
+                msg += f"stratigraphy.{key}: 'name' is missing. \n"
+            elif not isinstance(item["name"], str):
+                msg += f"stratigraphy.{key}: 'name' must be a string.\n"
+
+            if "stratigraphic" not in item:
+                msg += f"stratigraphy.{key}: 'stratigraphic' is missing.\n"
+            elif not isinstance(item["stratigraphic"], bool):
+                msg += f"stratigraphy.{key}: 'stratigraphic' must be a boolean.\n"
+
+            if "alias" in item:
+                if not isinstance(item["alias"], list):
+                    msg += f"stratigraphy.{key}: 'alias' must be a list.\n"
+                else:
+                    for alias in item["alias"]:
+                        if not isinstance(alias, str):
+                            msg += (
+                                f"stratigraphy.{key}: 'alias' items must be strings\n"
+                            )
+
+            if "stratigraphic_alias" in item:
+                if not isinstance(item["stratigraphic_alias"], list):
+                    msg += f"stratigraphy.{key}: 'stratigraphic_alias' must be list.\n"
+                else:
+                    for alias in item["stratigraphic_alias"]:
+                        if not isinstance(alias, str):
+                            msg += f"stratigraphy.{key}: 'stratigraphic_alias' items "
+                            msg += "must be strings.\n"
```

Comment:
Gets many line of codes. Could be an idea to refactor check of the global config to a new module (even a class)?

Reply:
Yes, agree. This doesn't scale very well. Have explored using a schema for this, and then using `jsonschema` to validate but experiments show it's not a great solution (either). Pydantic is also a possibility.

Reply:
I discussed this a little bit also in #383. This sounds like a separate issue, perhaps, and a slightly bigger job. But also sounds that it makes sense to carve out the functionality of the global_variables. Perhaps the config should be a class, rather than a dictionary inside fmu-dataio 🤷